src/compiler/crystal/interpreter/value.cr

```diff
@@ -67,6 +67,15 @@ struct Crystal::Repl::Value
     end
   end
 
+  def runtime_type : Crystal::Type
+    if type.is_a?(Crystal::UnionType)
```

Comment:
Also refer to [the `class` primitive's implementation in the interpreter](https://github.com/crystal-lang/crystal/blob/316422fb800826994c4623edb837f57b74f7a361/src/compiler/crystal/interpreter/primitives.cr#L89-L109)

Reply:
It should definitely match `value.class`. What would be the preferred way to have a single source of truth for that @HertzDevil ?

Reply:
For now just cross-reference the two places in comments, since there are probably even more places than those two that need to agree on the ABI (normal codegen and maybe stdlib)